htdocs/core/actions_massactions.inc.php

```diff
@@ -728,13 +728,14 @@
 	}
 
 	$arrayofinclusion=array();
-	foreach($listofobjectref as $tmppdf) $arrayofinclusion[]='^'.preg_quote($tmppdf.'.pdf','/').'$';
+	foreach($listofobjectref as $tmppdf) $arrayofinclusion[]='^'.preg_quote(preg_replace('#/#','_',$tmppdf).'.pdf','/').'$';
 	$listoffiles = dol_dir_list($uploaddir,'all',1,implode('|',$arrayofinclusion),'\.meta$|\.png','date',SORT_DESC,0,true);
 
 	// build list of files with full path
 	$files = array();
 	foreach($listofobjectref as $basename)
 	{
+		$basename = preg_replace('#/#', '_', $basename);
```

Comment:
What do you think of using 
dol_sanitizeFileName() instead of the preg_replace to replace / ?